hyperbolic-rs/preprocess.py

```diff
@@ -25,25 +25,28 @@
-                                                          '"gem" (keen-gem interactions), "ml-1m", '
-                                                          '"amzn-musicins", "amzn-vgames"')
-flags.DEFINE_string('dataset_path', default='data/amazon', help='Path to raw dataset: data/keen, data/ml-1m, '
-                                                               'data/amazon')
-flags.DEFINE_string('item_item_file', default='data/prep/amazon/musicins_musicins_cosine_distance_th0.6.pickle',
+flags.DEFINE_string('prep_id', default='foobar', help='Name of prep to store')
+flags.DEFINE_string('item', default='amazon', help='Item to process: "keen", "gem", "ml-1m" or "amazon"')
+flags.DEFINE_string('dataset_path', default='data/amazon', help='Path to raw dataset')
+flags.DEFINE_string('amazon_reviews', default='Musical_Instruments_5.json.gz',
+                    help='Name of the 5-core amazon reviews file')
+flags.DEFINE_string('amazon_meta', default='meta_Musical_Instruments.json.gz',
+                    help='Name of the 5-core amazon reviews file')
+flags.DEFINE_string('item_item_file', default='Musical_Instruments_Musical_Instruments_cosdist_th0.6.pickle',
                     help='Path to the item-item distance file')
 flags.DEFINE_boolean('plot_graph', default=False, help='Plots the user-item graph')
 flags.DEFINE_boolean('shuffle', default=False, help='Whether to shuffle the interactions or not')
-flags.DEFINE_boolean('add_extra_relations', default=True, help='For the amazon dataset, adds co-buy, co-view and'
-                                                                'categorical similarity relations')
-flags.DEFINE_integer('min_user_interactions', default=5, help='Users with less than this interactions are filtered')
-flags.DEFINE_integer('min_item_interactions', default=2, help='Items with less than this interactions are filtered')
-flags.DEFINE_integer('max_item_interactions', default=150, help='Items with more than this interactions are filtered')
+flags.DEFINE_boolean('add_extra_relations', default=True, help='For the amazon dataset, adds extra relations')
+flags.DEFINE_integer('min_user_interactions', default=5,
+                     help='Keens users with less than min_user_interactions are filtered')
+flags.DEFINE_integer('min_item_interactions', default=2,
+                     help='Keens/gems with less than this interactions are filtered')
+flags.DEFINE_integer('max_item_interactions', default=150,
+                     help='Keens/gems with more than this interactions are filtered')
 flags.DEFINE_integer('similarity_items_per_item', default=10, help='Amount of similarity items to add per item')
 flags.DEFINE_integer('seed', default=42, help='Random seed')
 flags.DEFINE_integer('filter_most_popular', default=-1,
-                     help='Filters out most popular items. If -1 it does not filter')
+                     help='Filters out most popular keens/gems. If -1 it does not filter')
```

Comment:
What happens if you try to combine this with other (non-keen) item types?

Reply:
It is not implemented in that case.
Other datasets are provided "cleaner", but in the case of Keen, we had to denoise/clean it a bit before using it.